### PR TITLE
VDS-242,VDS-243,VDS-244: Fix bugs in segment rule

### PR DESCRIPTION
**Pull request by asvishnyakov**

### CODE REVIEW

**Review by trueboroda — DISMISSED** [review]

**Review by ilyawzrd — DISMISSED** [review]

File: src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/customerSegments-preview.js [hunk 1]
```diff
@@ -1,93 +1,77 @@
...
-            take: 1000
+        // ui-grid
+        $scope.setGridOptions = function (gridId, gridOptions) {
+            $scope.gridOptions = gridOptions;
+            gridOptionExtension.tryExtendGridOptions(gridId, gridOptions);
+
+            gridOptions.onRegisterApi = function (gridApi) {
+                $scope.gridApi = gridApi;
+                gridApi.core.on.sortChanged($scope, function () {
+                    if (!blade.isLoading) blade.refresh();
```
Comment (pushnitsa) [inline src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/customerSegments-preview.js:61]: Add braces

File: src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/customerSegment-detail.js [hunk 2]
```diff
@@ -1,34 +1,32 @@
...
+        blade.isLoading = false;
+        blade.activeBladeId = null;
+
+        blade.currentEntity = {};
+
+        $scope.canSave = () => {
+            return false;
+        };
+
+        $scope.mainParameters = function () {
```
Comment (pushnitsa) [inline src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/customerSegment-detail.js:15]: Can we unified all methods to arrow functions?

File: src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/customerSegment-detail.js [hunk 3]
```diff
@@ -37,22 +35,25 @@ angular.module('virtoCommerce.DemoSolutionFeaturesModule')
...
                 };
                 blade.activeBladeId = ruleCreationBlade.id;
                 bladeNavigationService.showBlade(ruleCreationBlade, blade);
             };
 
-            $scope.$watch('blade.currentEntity', (data) => {
-                if (data) {
-                    $scope.totalPropertiesCount = 4;
-                    $scope.filledPropertiesCount = 0;
+        $scope.$watch('blade.currentEntity', (data) => {
```
Comment (pushnitsa) [inline src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/customerSegment-detail.js:47]: Here arrow function could be initialized without round brackets

**Review by pushnitsa — CHANGES_REQUESTED** [review]

File: src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/customerSegment-detail.js (hunk 3 above)
Comment (trueboroda) [reply src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/customerSegment-detail.js:47]: May be better to make these changes in the new customer segments module?

File: src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/customerSegments-preview.js (hunk 1 above)
Comment (asvishnyakov) [reply src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Scripts/blades/customerSegments-preview.js]: Fixed

**Pull request closed without merging**